torch/_inductor/codecache.py

```diff
@@ -761,25 +761,55 @@ def get_include_and_linking_paths(
         ipaths = cpp_extension.include_paths(cuda) + [sysconfig.get_path("include")]
         lpaths = []
         if sys.platform == "darwin":
-            # GNU OpenMP generally is not available on MacOS
-            # There is either Intel OpenMP(for x86) or LLVM OpenMP (for both x86 and arm64)
-            libs = ["omp"]
-            if os.getenv("OMP_PREFIX") is not None:
-                # Support OpenMP on MacOS
-                ipaths.append(os.path.join(os.getenv("OMP_PREFIX"), "include"))
-                lpaths.append(os.path.join(os.getenv("OMP_PREFIX"), "lib"))
-
-            if os.getenv("CONDA_PREFIX") is not None:
-                # On MacOS OpenMP is not available via the system install
-                # But on conda can be provided using https://anaconda.org/anaconda/llvm-openmp
-                conda_lib_path = os.path.join(os.getenv("CONDA_PREFIX"), "lib")
-                ipaths.append(os.path.join(os.getenv("CONDA_PREFIX"), "include"))
-                lpaths.append(conda_lib_path)
-                # Prefer Intel OpenMP on x86 machine
-                if os.uname().machine == "x86_64" and os.path.exists(
-                    os.path.join(conda_lib_path, "libiomp5.dylib")
-                ):
-                    libs = ["iomp5"]
+            cxx = cpp_compiler()
+            version_string = subprocess.check_output([cxx, "--version"]).decode('utf8')
```

Comment:
A subprocess is expensive, so we should cache this call (with `functools.lru_cache(None)`), so it only gets called once rather than every time.

Reply:
Fixed in [78ac245](https://github.com/pytorch/pytorch/pull/107111/commits/78ac245f0ee34d0647972db4f212c8bff4f8e459).